src/coreclr/nativeaot/System.Private.CoreLib/src/System/CrashInfo.cs

```diff
@@ -0,0 +1,389 @@
+                RhFailFastReason.InternalError or
+                RhFailFastReason.ClassLibDidNotTranslateExceptionID => CrashReason.InternalFailFast,
+                RhFailFastReason.UnhandledException or
+                RhFailFastReason.UnhandledExceptionFromPInvoke or
+                RhFailFastReason.UnhandledException_ExceptionDispatchNotAllowed or
+                RhFailFastReason.UnhandledException_CallerDidNotHandle => CrashReason.UnhandledException,
+                _ => CrashReason.Unknown,
+            };
+
+            if (!WriteValue("reason", (int)crashReason))
+                return false;
+
+            if (!WriteValue("message", message, max: 1024))
+                return false;
+
+            return true;
+        }
+
+        /// <summary>
+        /// Adds the exception info to the JSON buffer under the "exception" key. If the exception can not fit in
+        /// the triage buffer the allocations are backed out.
+        /// </summary>
+        /// <param name="key">the name of the block</param>
+        /// <param name="exception">exception build triage block from</param>
+        /// <param name="maxMessageSize">limits the size of the exception message strings</param>
+        /// <param name="maxNumberStackFrames">limits the number of stack frames written to the triage buffer</param>
+        /// <param name="maxMethodNameSize">limits the size of the stack frame method name strings</param>
+        /// <returns>true - success, false - out of triage buffer space</returns>
+        private bool WriteExceptionWithFallback(ReadOnlySpan<char> key, Exception exception, int maxMessageSize, int maxNumberStackFrames, int maxMethodNameSize)
+        {
+            int savedIndex = _currentBufferIndex;
+            if (!WriteException(key, exception, maxMessageSize, maxNumberStackFrames, maxMethodNameSize))
+            {
+                _currentBufferIndex = savedIndex;
+                return false;
+            }
+            return true;
+        }
+
+        /// <summary>
+        /// Adds the exception info to the JSON buffer
+        /// </summary>
+        /// <param name="key">the name of the block</param>
+        /// <param name="exception">exception build triage block from</param>
+        /// <param name="maxMessageSize">limits the size of the exception message strings</param>
+        /// <param name="maxNumberStackFrames">limits the number of stack frames written to the triage buffer</param>
+        /// <param name="maxMethodNameSize">limits the size of the stack frame method name strings</param>
+        /// <returns>true - success, false - out of triage buffer space</returns>
+        private bool WriteException(ReadOnlySpan<char> key, Exception exception, int maxMessageSize, int maxNumberStackFrames, int maxMethodNameSize)
+        {
+            if (!OpenValue(key, '{'))
+                return false;
+
+            if (!WriteHexValue("address", (ulong)Unsafe.AsPointer(ref exception)))
```

Comment:
All these keys are string that you know need no escaping and no casting. Define them as u8 strings and just pass them down.

Reply:
I like that idea that all the keys are u8.  I make that change.

Reply:
Is there a UTF8 literal char? That I can replace the ", {, }, etc. and write the byte directly.